tmt/utils.py

```diff
@@ -2045,9 +2068,7 @@ def run(command: str) -> str:
 
     # Get the ref (skip for the default)
     ref = run('git rev-parse --abbrev-ref HEAD')
-    def_branch = default_branch(git_root)
-    if ref != def_branch:
-        fmf_id.ref = ref
+    fmf_id.ref = ref
```

Comment:
So this means that the fmf identifier will always include the `ref` key, even for the default branch. This does not hurt, although on the other hand it means that the short/concise form will never be used here. And yet from another point of view, always including the `ref` value means to be rather explicit than implicit :)

@lukaszachy, @thrix, @happz, @FrNecas, any thoughts on this? Are we ok with the change?

Reply:
I prefer explicit, on the other hand, I have no idea what might this tiny change cause :) Seems safe & sane enough to me.

Reply:
How about removing 'ref' in the nitrate export? https://github.com/teemtee/tmt/pull/1727/files#diff-42b8602016be38f13b3821a4727ede1acbeb68fcc0aedf1928e37a48a49730beL638

Reply:
> The always present 'ref' will break current 'use shallow clone when no ref is specified' workarounds to spare bandwith...

Ah, good point!

> Thus +1 but it will take some time to get wow/wrapper updated.

So, I guess, we should probably wait a bit until the changes are in place to prevent putting unnecessary load on the git server...

Reply:
> How about removing 'ref' in the nitrate export?

This means that we would have to do the default branch check here instead of the original place. Thinking about whether it is worth to move check from one place to another...

Reply:
How about changing `fmf_id` to accept option to always fill the 'ref' (default to false)? `public_web_url` will ask for it, other calls not.

Reply:
Implemented, also changed what is passed to the public_web_url function, now sending the Test instead of just FmfId, probably easier now if we wanted to use it in other places.

Reply:
Well, I completely agree and would very much prefer to skip the last commit. Because that was reasonable number of arguments and reasonable content of the function.
However when I can't rely on test.fmf_id to have the ref and having to call the fmf_id function explicitly (can't even do it with test.fmf_id as that is a property), then I prefer to do that inside the function and not force everyone that ever calls the public_web_url to know that he has to call fmf_id function himself.

Reply:
> I liked the previous incarnation, a public URL derived from (any) fmf id...

Agree with @happz here. The `public_web_url()` dedicated for tests only does not sound right.

> However when I can't rely on test.fmf_id to have the ref and having to call the fmf_id function explicitly (can't even do it with test.fmf_id as that is a property),

Hm, right, here a `Test.fmf_id()` method would be more flexible and useful rather than a property. Maybe a general hint not to use properties too much?

Thinking about it a bit more... what about this:

 * `tmt.utils.clickable_git_url` takes `url`, `ref` and optional `path` and returns a clickable link (reusable for other use cases as well)
 * a new method `Core.clickable_url()` is introduced which will take care of feeding `public_web_url` with the right parameters

What do you think?

Reply:
Yes, I believe the solution for this should be relatively easy: Feed `tmt.utils.clickable()` with fmf id path plus path to the directory where the metadata are stored as we do for the `Test.path` default:

https://github.com/teemtee/tmt/blob/2dfbbb4e488b0589c0fc0652b4ed3287fa3be50e/tmt/base.py#L707-L722

Reply:
Maybe like this? Core now has clickable_url with optional path argument and is calling fmf_id function to get all the args and returns the link. public_web_url now accepts url and ref as strings and optional path.

Reply:
This looks much better!